.github/workflows/registry_upload.yml

```diff
@@ -0,0 +1,78 @@
+name: Create and publish a Docker image
```

Comment:
I think we want to merge the workflow with the other workflow. So that the publishing of docker images only happens when every test passes.

Reply:
The way I want it set up is that only after the PR is merged, the images are published. And all the test will have to pass before the PR can be merged.

Reply:
I updated the trigger to run after CI is completed. I also added a test to only build of CI succeeded.